React-Todo-list-main/src/App.js

```diff
@@ -19,25 +15,58 @@ function App() {
   /**
    * initialzing user
    */
-  const user = {
-    userID: "user-us",
-    email: "upendra.singh@ltts.com",
-  };
+
+  const [user, setUser] = useState(JSON.parse(localStorage.getItem("user")));
+
   console.log("User: ", user);
 
+  const [isLoggedIn, setIsLoggedIn] = useState(
+    localStorage.getItem("isLoggedIn") === "true"
+  );
+
+  /**
+   * Setting the user and flag from the login details
+   * @param {*} userName
+   */
+  const setUserDetails = (userName) => {
+    let localUser = {
+      userID: userName,
+      email: "upendra.singh@ltts.com",
+    };
+
+    localStorage.setItem("user", JSON.stringify(localUser));
+    localStorage.setItem("isLoggedIn", true);
+    setUser(localUser);
+    setIsLoggedIn(true);
+  };
+
+  /**
+   * logging out from todo app
+   */
+  const onLogout = () => {
+    localStorage.setItem("user", JSON.stringify({}));
+    localStorage.setItem("isLoggedIn", false);
+    setUser({});
+    setIsLoggedIn(false);
+  };
+
   return (
-    <StatsigProvider
-      sdkKey={SS_CLIENT_SDK_KEY}
-      waitForInitialization={true}
-      options={{
-        environment: { tier: "staging" },
-      }}
-      user={user}
-    >
-      <div className="App">
-        <TodoWrapper />
-      </div>
-    </StatsigProvider>
+    <div className="App">
+      {!isLoggedIn ? (
+        <Login setUser={setUserDetails} />
+      ) : (
```

Comment:
If the user is not logged in, still render the `StatsigProvider` - it just wont have a logged in user.  But youve handled the user state just fine:

<div className="App">
  <StatsigProvider
          sdkKey={CLIENT_SDK_KEY}
          waitForInitialization={true}
          options={{
            environment: { tier: "staging" },
          }}
          user={user}
        >
{!isLoggedIn ? (
        <Login setUser={setUserDetails} />
      ) : <TodoWrapper onLogout = {onLogout} />
</StatsigProvider>
```

Reply:
In this are case are initializing the provider twice, once without user and once with user?